DuckDuckGo/Subscription/PrivacyProDataReporting.swift

```diff
@@ -0,0 +1,310 @@
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import BrowserServicesKit
+import DDGSync
+
+enum PrivacyProPromoParameters: String, CaseIterable {
+    case isReinstall
+    case fireButtonUser = "fireButtonUsed"
+    case syncUsed
+    case fireproofingUsed
+    case appOnboardingCompleted
+    case emailEnabled
+    case widgetAdded
+    case frequentUser
+    case longTermUser
+    case autofillUser
+    case validOpenTabsCount
+    case searchUser
+
+    /// Pick a randomized subset of parameters each time they are attached to a pixel
+    static func randomizedSubset() -> [PrivacyProPromoParameters] {
+        Array(allCases.shuffled().prefix(Int(allCases.count * 2/3)))
+    }
+}
+
+protocol PrivacyProDataReporting {
+    func isReinstall() -> Bool
+    func isFireButtonUser() -> Bool
+    func isSyncUsed() -> Bool
+    func isFireproofingUsed() -> Bool
+    func isAppOnboardingCompleted() -> Bool
+    func isEmailEnabled() -> Bool
+    func isWidgetAdded() -> Bool
+    func isFrequentUser() -> Bool
+    func isLongTermUser() -> Bool
+    func isAutofillUser() -> Bool
+    func isValidOpenTabsCount() -> Bool
+    func isSearchUser() -> Bool
+}
+
+// swiftlint:disable identifier_name
+final class DefaultPrivacyProDataReporter: PrivacyProDataReporting {
+    enum Key {
+        static let fireCountKey = "com.duckduckgo.ios.privacypropromo.FireCount"
+        static let isFireproofingUsedKey = "com.duckduckgo.ios.privacypropromo.FireproofingUsed"
+        static let isWidgetAddedKey = "com.duckduckgo.ios.privacypropromo.WidgetAdded"
+        static let applicationLastSessionEndedKey = "com.duckduckgo.ios.privacypropromo.ApplicationLastSessionEnded"
+        static let searchCountKey = "com.duckduckgo.ios.privacypropromo.SearchCount"
+    }
+
+    enum Constants {
+        static let includedOrigins = "origins"
+    }
+
+    enum UseCase {
+        case messageID(String)
+        case origin(String?)
+        case debug
+    }
+
+    public static let shared = DefaultPrivacyProDataReporter()
+
+    private static let fireCountThreshold = 5
+    private static let frequentUserThreshold = 2
+    private static let longTermUserThreshold = 30
+    private static let autofillUserThreshold = 5
+    private static let openTabsCountThreshold = 3
+    private static let searchCountThreshold = 50
+
+    private lazy var includedOrigins = config.settings(for: .additionalCampaignPixelParams)[Constants.includedOrigins] as? [String]
+
+    private let config: PrivacyConfiguration
+    private let variantManager: VariantManager
+    private let userDefaults: UserDefaults
+    private let emailManager: EmailManager
+    private let tutorialSettings: TutorialSettings
+    private let appSettings: AppSettings
+    private let statisticsStore: StatisticsStore
+    private let secureVault: (any AutofillSecureVault)?
+    private var syncService: DDGSyncing?
+    private var tabsModel: TabsModel?
+    private let dateGenerator: () -> Date
```

Comment:
`syncService` and `tabsModel` are the trickiest to initialize, so I chose to inject them at runtime.